src/ussd_pmtct_rapidpro.js

```diff
@@ -1,24 +1,413 @@
+                    if(opts.http_error_count === 3) {
+                        self.im.log.error(e.message);
+                        return self.states.create("__error__");
+                    }
+                    return self.states.create(name, opts);
+                });
+        });
+
+        self.states.add("state_optout", function(name) {
+            return new ChoiceState(name, {
+                question: "The mother is currently receiving messages about keeping her baby " +
+                            "HIV-negative. Does she want to stop getting these messages?",
+                error:
+                    "Sorry, please reply with the number next to your answer. " +
+                    "Does she want to stop getting these messages?",
+                choices: [
+                    new Choice("yes", "Yes"),
+                    new Choice("no", "No")
+                ],
+                next: function(choice) {
+                    if(choice.value === "yes") {
+                        return "state_optout_reason";
+                    } else {
+                        return "state_no_optout";
+                    }
+                }
+            });
+        });
+
+        self.states.add("state_no_optout", function(name) {
             return new EndState(name, {
-                text: $(
-                    "You need to be registered on MomConnect to receive these messages. Please visit the nearest " +
-                    "clinic to register."
+                next: "state_start",
+                text:
+                    "Thanks! MomConnect will continue to send HER helpful messages and process your " +
+                    "personal info."
+            });
+        });
+
+        self.add("state_optout_reason", function(name) {
+            var question = "Please tell us why she no longer wants to get msgs:";
+            return new PaginatedChoiceState(name, {
+                question: question,
+                error: question,
+                accept_labels: true,
+                options_per_page: null,
+                characters_per_page: 160,
+                choices: [
+                    new Choice("not_hiv_positive", "She's not HIV+"),
+                    new Choice("miscarriage", "Miscarriage"),
+                    new Choice("stillborn", "Baby was stillborn"),
+                    new Choice("babyloss", "Baby passed away"),
+                    new Choice("not_useful", "Msgs aren't helpful"),
+                    new Choice("other", "Other"),
+                    new Choice("unknown", "I prefer not to say")
+                ],
+                next: function(choice) {
+                    self.im.user.set_answer("optout_reason", choice.value);
+                    if(_.includes(["miscarriage", "stillborn", "babyloss"], choice.value)) {
+                        return "state_loss_optout";
+                    } else {
+                        return "state_opted_out";
```

Comment:
We need to be submitting the optout to RapidPro here, not going to the end state

Reply:
ahh yes, thanks for picking this up, forgot I changed a few things around